src/rdkafka.c

```diff
@@ -270,7 +279,463 @@ void rd_kafka_log0 (const rd_kafka_conf_t *conf,
+                                        char *end_ptr;
+                                        life_seconds_long = strtoll(life_seconds_text, &end_ptr, 10);
+                                        if (*end_ptr != '\0') {
+                                                rd_snprintf(errstr, errstr_size,
+                                                        "Invalid sasl.oauthbearer.config: non-integral '%s': %s",
+                                                        prefix_life_seconds, life_seconds_text);
+                                        } else if (life_seconds_long <= 0 || life_seconds_long > INT_MAX) {
+                                                rd_snprintf(errstr, errstr_size,
+                                                        "Invalid sasl.oauthbearer.config: value out of range of positive int '%s': %s",
+                                                        prefix_life_seconds, life_seconds_text);
+                                        } else {
+                                                *life_seconds = life_seconds_long;
+                                        }
+                                }
+                        }
+                        rd_free(life_seconds_text);
+                } else if (!strncmp(prefix_extension, loc, strlen(prefix_extension))) {
+                        char *extension_key = NULL;
+                        parse_unsecured_jws_config_value_for_prefix(&loc, prefix_extension, '=',
+                                &extension_key, errstr, errstr_size);
+                        if (*errstr == '\0') {
+                                if (*extension_key == '\0') {
+                                        rd_snprintf(errstr, errstr_size,
+                                                "Invalid sasl.oauthbearer.config: empty '%s' key",
+                                                prefix_extension);
+                                } else {
+                                        char *extension_value = NULL;
+                                        parse_unsecured_jws_config_value_for_prefix(&loc, "", ' ',
+                                                &extension_value, errstr, errstr_size);
+                                        if (*errstr == '\0') {
+                                                rd_list_add(extensions,
+                                                        rd_strtup_new(extension_key, extension_value));
+                                                rd_free(extension_value);
+                                        }
+                                }
+                                rd_free(extension_key);
+                        }
+                } else {
+                        rd_snprintf(errstr, errstr_size,
+                                "Unrecognized sasl.oauthbearer.config beginning at: %s", loc);
+                }
+        }
+
+        rd_free(cfg_copy);
 
+        return *errstr == '\0' ? 0 : -1;
```

Comment:
Don't use errstr as error indication, just error string propagating. Use an int to keep track if there was an error, or goto..

Reply:
👍